### PR TITLE
Add a help page

### LINKED ISSUE #1: support help command
e.g. "create: help"

### PR DESCRIPTION
**Pull request by chriskuehl**
Fixes #1

Running at http://supernova.ocf.berkeley.edu:8888/ (will leave this up for a while)

The page is pretty rough right now, but I'm all about shipping something which is at least minimally useful and improving it later :)

The plan is to have the ircbot itself serve this page (the code in this branch does that) and add configs to marathon/lb nginx to route ircbot.o.b.e there.

### CODE REVIEW

File: ircbot/plugin/help.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                self.wfile.write(b'404 File not found')
+            else:
+                plugins = collections.defaultdict(set)
+                for listener in bot.listeners:
+                    plugins[bot.plugins[listener.fn.__module__]].add(listener)
+                rendered = jinja_env.get_template('plugin/templates/help.html').render(
+                    plugins=sorted(plugins.items(), key=lambda p: p[0].__name__),
+                ).encode('utf8')
+
+                self.send_response(200, 'Okay')
```
Comment (kpengboy) [inline ircbot/plugin/help.py]: I would use the standard response message...

File: ircbot/plugin/help.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                self.send_header('Content-Length', len(rendered))
+                self.send_header('Content-Type', 'text/html; charset=utf-8')
+                self.end_headers()
+                self.wfile.write(rendered)
+
```
Comment (kpengboy) [inline ircbot/plugin/help.py]: Might want to consider supporting the HEAD method as well? I think many clients expect that method to exist

File: ircbot/plugin/templates/help.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    <head>
+        <title>ocf ircbot help</title>
+        <style>
+            tt { background-color: #eee; padding: 3px; }
+        </style>
+    </head>
+    <body>
+        <h1>create: <em>an IRC bot for doing stupid stuff and sometimes handling commands for account creation</em></h1>
+        <hr />
+
```
Comment (kpengboy) [inline ircbot/plugin/templates/help.html]: I would add a line like "Available commands are:" here for context

File: ircbot/plugin/help.py (hunk 1 above)
Comment (chriskuehl) [reply ircbot/plugin/help.py]: why?

File: ircbot/plugin/help.py (hunk 2 above)
Comment (chriskuehl) [reply ircbot/plugin/help.py]: HEAD is probably a nice-to-have. I've done a bunch of smaller http servers in this pattern before and no clients have had an issue with it. Feel free to add it though.

File: ircbot/plugin/templates/help.html (hunk 3 above)
Comment (chriskuehl) [reply ircbot/plugin/templates/help.html]: added. though i suspect we might change all of this later anyway if we put effort into making the page look better.

**Comment by chriskuehl** [comment]
Another problem is that, as-written, we won't be able to run two dev ircbots on one host because they'll both want to bind to this port. Not sure what the best way to handle that is yet.

**Pull request merged**